test/download.spec.js

```diff
@@ -70,6 +70,21 @@ describe('Download', function() {
     expect(fs.readFileSync(path).toString()).toBe('Hello world');
     await page.close();
   });
+  it(`should report download path within page.on('download', …) handler`, async({browser, server}) => {
+    const page = await browser.newPage({ acceptDownloads: true });
+    const onDownloadPathPath = new Promise((res, rej) => {
+      setTimeout(() => { rej('failed to find path quickly'); }, 1000);
```

Comment:
This will flake on the bots, you should rather await dl.path() and let it time out.